### PR TITLE
Tidal power performance

### PR DESCRIPTION
**Pull request by jmcvey3**
Tidal power performance capabilities specified in IEC/TS 6200-200.

TODOs left:
- [x] Add "assertions" to input arguments
- [x] Add tests
- [x] Flesh out jupyter notebook example
- [x] Extend profiles to seafloor and surface

I would appreciate a second set of eyes and would be happy to discuss interpretation of the standards. The mathematical section is not well written, and I was relying on information conveyed in the text and plots in the appendix to figure out what was being asked. (My TS copy is heavily loaded with comments and corrections to keep track of things).

Also includes bugfixes, docstring clarifications, and useability edits for the Tidal module in general that I came across from a users standpoint. Changelog as follows:

   - Bugfixes: 
            - Standardized function inputs to accept both numpy.ndarrays and pandas.Series (some functions already have this implemented).
            - Fixed assertions so that if a numpy array is supplied, the following assertions that rely on pandas don't fail (some functions already have this implemented).
            - Added a (temporary) fix so that assertions can handle nans 
            - Fixed `river.performance` capture area calculation the multiplies by instead of dividing by 4.
            - Fixed a misnamed variable in `tidal.performance.principal_flow_directions`.
            - Corrected the "Returns" statement in the `tidal.performance.principal_flow_directions` docstring and added a "Notes" section.
           
   - API/useability
            - Added 'ax' argument passthrough to `tidal.graphics.plot_rose` and `tidal.graphics.plot_joint_prob_distribution` to enable more complex matplotlib capabilities outside of the function.
            - Added check and updated docstring in `tidal.performance.principal_flow_directions` to make sure directions are between 0 and 360 degrees, which the function requires.

### CODE REVIEW

**Comment by jmcvey3** [comment]
@ssolson This should be ready for review 

Also, the tests are failing because of something I thought it an error in `river.performance.py` capture area functions: the area of a circle is 1/4 pi d^2, not sure if the original 4 pi d^2 is an error or not

**Comment by jmcvey3** [comment]
@ssolson I just remembered there are some changes in the dolfyn PR (namely to the function/shortcut `dolfyn.velocity.U_dir`) that will affect this one. I'll try to wrap the dolfyn PR soon here so we can merge these back to back.

File: mhkit/tidal/performance.py
Comment (ssolson) [inline mhkit/tidal/performance.py:1]: I created a PR to address the change needed for this file.

**Review by ssolson — COMMENTED** [review]
James I still need to review the example notebooks which I will get to later next week.

File: mhkit/tidal/resource.py [hunk 1]
```diff
@@ -75,21 +75,27 @@ def principal_flow_directions(directions, width_dir):
...
+    One must determine which principal direction is flood and which is 
+    ebb based on knowledge of the measurement site.
     '''
 
-    if isinstance(directions,np.ndarray) ==1:
-        directions=pd.Series(directions) 
+    if isinstance(directions, np.ndarray):
+        directions=pd.Series(directions)
+    assert(all(directions>=0) and all(directions<=360),
+           'flood must be between 0 and 360 degrees')
```
[the remark this answers is not in the document]
Comment (jmcvey3) [reply mhkit/tidal/resource.py:97]: Oh good! I was trying to be consistent

File: mhkit/tidal/performance.py
Comment (jmcvey3) [reply mhkit/tidal/performance.py:1]: Merged

**Comment by jmcvey3** [comment]
@ssolson This one should also be good to merge in if you want to give it one more review

**Comment by ssolson** [comment]
@jmcvey3 merging #232 created a conflict with the adcp example. 

I created PR https://github.com/jmcvey3/MHKiT-Python/pull/5 on your branch which resolves the issue.

**Comment by jmcvey3** [comment]
Thanks @ssolson - made some last fixes to the notebooks

**Comment by ssolson** [comment]
All tests pass excluding hindcast. Hindcast functionality was not changed in this PR.
image

**Pull request merged**